discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -76,6 +78,28 @@ struct IngestedDocument {
     properties: DocumentProperties,
 }
 
+#[derive(Debug, Clone, Serialize)]
+struct IngestionError {
+    /// List of Document Indices which were not successfully processed
+    documents: Vec<ErroredDocumentId>,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct ErroredDocumentId {
+    id: DocumentId,
+}
+
+impl IngestionError {
```

Comment:
we could also add `to_reply` method similar to `PersonalizedDocumentsError` in user API

```rust
pub(crate) fn to_reply(&self) -> impl Reply {
    reply::with_status(reply::json(self), StatusCode::INTERNAL_SERVER_ERROR)
}
```

Reply:
I added it here, but I had removed it before push as it was marked by `cargo check` as a dead code

Reply:
it won't be dead code if you use it in `handle_rejection` below :)

Reply:
Okay, I will get it back then